slackbridge/messages.py

```diff
@@ -71,19 +83,66 @@ def resolve(self):
                 user_bot.leave(channel_name)
             return
 
+    def is_bot_user(self):
+        """Sometimes bot_id is not included and other
+        times it is passed as None. Checks both cases."""
+        return ('bot_id' in self.raw_message and
+                self.raw_message['bot_id'] is not None)
+
     def _change_presence(self, user_bot):
         if self.raw_message['presence'] == 'away':
             user_bot.away('Slack user inactive.')
         elif self.raw_message['presence'] == 'active':
             user_bot.back()
 
-    def _irc_me_action(self, channel_name, user_bot):
+    def _irc_me_action(self, channel_name, user_bot, action):
         user_bot.post_to_irc(
             user_bot.describe,
             '#' + channel_name,
-            self.raw_message['text'],
+            action,
         )
 
+    def _post_to_fluffy(self, channel_name, user_bot, file_data):
+        ext = file_data['filetype']
+        # Adapted from https://api.slack.com/tutorials/working-with-files
+        auth = {'Authorization': 'Bearer {}'.format(
+            self.bridge_bot.slack_token
+        )}
+        r = requests.get(
+            file_data['url_private'],
+            headers=auth,
+            stream=True,
+        )
+        if r.status_code != 200:
+            log.err('Could not GET image from: {}'.format(
+                file_data['url_private'],
+            ))
+            return
+
+        with NamedTemporaryFile(suffix='.' + ext) as tf:
+            # Download and put file in /tmp
+            r.raw.decode_content = True
+            shutil.copyfileobj(r.raw, tf)
+
+            # Upload File to fluffy
+            tf.seek(0)
+            r = requests.post(
+                FILEHOST + '/upload',
+                files={'file': tf},
+                allow_redirects=False,
+            )
+            if r.status_code not in (301, 302):
+                log.err('Failed to upload (status code {}):'.format(
+                    r.status_code,
+                ))
+                return
+
+            self._irc_me_action(
+                channel_name,
+                user_bot,
+                'uploaded an image: ' + r.headers['Location'],
```

Comment:
I'd change this so it's not just `an image`, because this could be any file at this point.